backend/cellguide/pipeline/ontology_tree/tree_builder.py

```diff
@@ -277,52 +279,75 @@ def get_ontology_tree_state_per_celltype(self) -> Dict[str, OntologyTreeState]:
              - "isExpandedNodes": A list of cell type ontology term ids that are expanded by default.
              - "notShownWhenExpandedNodes": A dictionary that maps cell type ontology term ids to their hidden children
         """
-        all_states_per_cell_type = {}
-        for i, end_node in enumerate(self.all_cell_type_ids_in_corpus):
-            if end_node in self.traverse_node_counter:
-                logger.info(
-                    "Getting ontology tree state for cell type %s (%s/%s)",
-                    end_node,
-                    i + 1,
-                    len(self.all_cell_type_ids_in_corpus),
-                )
+        results = {
+            celltype: delayed(self._process_cell_type__parallel)(celltype)
+            for celltype in self.all_cell_type_ids_in_corpus
+            if celltype in self.traverse_node_counter
+        }
+
+        logger.info(
+            f"Getting ontology tree states for {len(results)} cell types using {CELLGUIDE_PIPELINE_NUM_CPUS} CPUs..."
+        )
+        with ProgressBar():
+            computed_results = compute(*list(results.values()), num_workers=CELLGUIDE_PIPELINE_NUM_CPUS)
+            all_states_per_cell_type = {
+                cell_type_id: result for cell_type_id, result in zip(results.keys(), computed_results)
+            }
 
-                all_paths = []
-                siblings = []
-                for i in range(self.traverse_node_counter[end_node]):
-                    end_node_i = end_node + "__" + str(i)
-                    path = self._depth_first_search_pathfinder(end_node_i)
-                    path = path if path else [end_node_i]
-                    all_paths.append(path)
+        return all_states_per_cell_type
 
-                    siblings.append(
-                        sum([self.all_children.get(parent, []) for parent in self.all_parents.get(end_node_i, [])], [])
-                    )
+    def _process_cell_type__parallel(self, celltype: str) -> OntologyTreeState:
```

Comment:
this function is the outer loop for processing the cell types and is parallelized